### PR TITLE
fix: Arg completions for case class in nested apply

### PR DESCRIPTION
**Pull request by jkciesluk**
fixes https://github.com/scalameta/metals/issues/5156

Sometimes on nested apply we get `method.tpe == null`, for which we have fallback way to get parameters. The cases were in wrong order, since for classes (and now case classes) there is a simpler way.

### CODE REVIEW

File: mtags/src/main/scala-2/scala/meta/internal/pc/completions/ArgCompletions.scala [hunk 1]
```diff
@@ -22,7 +22,13 @@ trait ArgCompletions { this: MetalsGlobal =>
     val method: Tree = typedTreeAt(funPos)
     val methodSym = method.symbol
     lazy val baseParams: List[Symbol] =
-      if (method.tpe == null) {
+      if (methodSym.isModule) {
+        methodSym
+          .localCompanion(funPos)
```
Comment (jkciesluk) [inline mtags/src/main/scala-2/scala/meta/internal/pc/completions/ArgCompletions.scala:27]: If the case class was defined inside a method, we can't use `methodSym.companionClass`

File: mtags/src/main/scala-2/scala/meta/internal/pc/PcCollector.scala [hunk 2]
```diff
@@ -73,20 +73,7 @@ abstract class PcCollector[T](
    * @param sym symbol to find a companion for
    * @return companion if it exists
    */
```
Comment (kasiaMarek) [inline mtags/src/main/scala-2/scala/meta/internal/pc/PcCollector.scala:75]: Leftover comment.

File: mtags/src/main/scala-2/scala/meta/internal/pc/completions/ArgCompletions.scala [hunk 3]
```diff
@@ -22,7 +22,13 @@ trait ArgCompletions { this: MetalsGlobal =>
     val method: Tree = typedTreeAt(funPos)
     val methodSym = method.symbol
     lazy val baseParams: List[Symbol] =
-      if (method.tpe == null) {
+      if (methodSym.isModule) {
```
Comment (kasiaMarek) [inline mtags/src/main/scala-2/scala/meta/internal/pc/completions/ArgCompletions.scala:25]: My understanding of this code is as follows:
if it's an object take the companion class and get args for its constructor.
What if we want arg completions for an `apply` defined in the companion object? Will that work?
```
case class A(val a: Int, val b: Int)
object A {
  def apply(a: Int): A = new A(a, 3)
}
```

**Review by kasiaMarek — CHANGES_REQUESTED** [review]

File: mtags/src/main/scala-2/scala/meta/internal/pc/completions/ArgCompletions.scala (hunk 3 above)
Comment (jkciesluk) [reply mtags/src/main/scala-2/scala/meta/internal/pc/completions/ArgCompletions.scala]: Thanks for noticing. I changed it so instead of using companion class, we search for `apply` method and we prefer the one defined in companion object.

**Comment by kasiaMarek** [comment]
That deserves a new issue but I feel like if we have overloaded methods we should not choose the first or last one of the matching methods to show args completions but rather collect args from all the matching methods, so for:
```
def m(inn : Int) = ???
def m(idd : Option[Int]) = ???
def k = m(I@@)
```
we'd get something like:
```
idd := Option[Int]
inn := Int
```
What do you reckon?

**Comment by jkciesluk** [comment]
> That deserves a new issue but I feel like if we have overloaded methods we should not choose the first or last one of the matching methods to show args completions but rather collect args from all the matching methods, so for:
> 
> ```
> def m(inn : Int) = ???
> def m(idd : Option[Int]) = ???
> def k = m(I@@)
> ```
> 
> we'd get something like:
> 
> ```
> idd := Option[Int]
> inn := Int
> ```
> 
> What do you reckon?

i totally agree that until we know exactly which method user is trying to use, we should suggest named arguments for every possibility. But let's do that in a different pr

**Comment by kasiaMarek** [comment]
> But let's do that in a different pr

Here's an issue about this: https://github.com/scalameta/metals/issues/5266

File: mtags/src/main/scala-2/scala/meta/internal/pc/completions/ArgCompletions.scala [hunk 4]
```diff
@@ -169,6 +170,18 @@ trait ArgCompletions { this: MetalsGlobal =>
       }
     }
 
+    private def getParamsFromObject(): List[Symbol] = {
+      methodSym.info.members
+        .collect {
+          case m
+              if m.decodedName == "apply" &&
+                m.paramss.flatten.length >= apply.args.length =>
+            m.paramss.flatten
```
Comment (tgodzik) [inline mtags/src/main/scala-2/scala/meta/internal/pc/completions/ArgCompletions.scala:179]: We can do it in another PR for sure, but we should probably also match it by names/types that already exist in the apply

**Review by tgodzik — APPROVED** [review]
Great work! We can improve further in next PRs

**Review by kasiaMarek — APPROVED** [review]
